ethereum_user_connector/src/Plugin/rest/resource/ValidateEthereumAccount.php

```diff
@@ -86,16 +86,19 @@ public static function create(ContainerInterface $container, array $configuratio
    *
    * @throws \Symfony\Component\HttpKernel\Exception\HttpException
    *   Throws exception expected.
+   * @throws \Exception
    *
-   * @return ResourceResponse
+   * @return \Drupal\rest\ResourceResponse
    *   With validated data.
    *   See ethereum_user_connector\Controller\verifyUserByHash().
    */
   public function get($hash) {
 
     if ($this->currentUser->isAuthenticated()) {
-      $eth = new EthereumUserConnectorController();
-      $validation = $eth->verifyUserByHash(Xss::filter($hash));
+
+      $controller = new EthereumUserConnectorController();
```

Comment:
Here and in all the other places which instantiate `EthereumUserConnectorController`, we need to do it through the class resolver instead, like this: `\Drupal::service('class_resolver')->getInstanceFromDefinition(EthereumUserConnectorController::class)`

This will properly inject the dependencies from the container, and I'm pretty sure it's the reason for the changes to the constructor of `\Drupal\ethereum\Controller\EthereumController` (i.e. those changes can be reverted)